packages/block-editor/src/components/block-list/block.js

```diff
@@ -49,108 +49,181 @@ import Inserter from '../inserter';
 import HoverArea from './hover-area';
 import { isInsideRootBlock } from '../../utils/dom';
 
-export class BlockListBlock extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.setBlockListRef = this.setBlockListRef.bind( this );
-		this.bindBlockNode = this.bindBlockNode.bind( this );
-		this.setAttributes = this.setAttributes.bind( this );
-		this.maybeHover = this.maybeHover.bind( this );
-		this.forceFocusedContextualToolbar = this.forceFocusedContextualToolbar.bind( this );
-		this.hideHoverEffects = this.hideHoverEffects.bind( this );
-		this.onFocus = this.onFocus.bind( this );
-		this.preventDrag = this.preventDrag.bind( this );
-		this.onPointerDown = this.onPointerDown.bind( this );
-		this.deleteOrInsertAfterWrapper = this.deleteOrInsertAfterWrapper.bind( this );
-		this.onBlockError = this.onBlockError.bind( this );
-		this.onTouchStart = this.onTouchStart.bind( this );
-		this.onClick = this.onClick.bind( this );
-		this.onDragStart = this.onDragStart.bind( this );
-		this.onDragEnd = this.onDragEnd.bind( this );
-		this.selectOnOpen = this.selectOnOpen.bind( this );
-		this.hadTouchStart = false;
-
-		this.state = {
-			error: null,
-			dragging: false,
-			isHovered: false,
-		};
-		this.isForcingContextualToolbar = false;
-	}
+/**
+ * Prevents default dragging behavior within a block to allow for multi-
+ * selection to take effect unhampered.
+ *
+ * @param {DragEvent} event Drag event.
+ *
+ * @return {void}
+ */
+const preventDrag = ( event ) => {
+	event.preventDefault();
+};
+
+function BlockListBlock( {
+	blockRef,
+	mode,
+	isFocusMode,
+	hasFixedToolbar,
+	isLocked,
+	isFirst,
+	isLast,
+	clientId,
+	rootClientId,
+	isSelected,
+	isPartOfMultiSelection,
+	isFirstMultiSelected,
+	isTypingWithinBlock,
+	isCaretWithinFormattedText,
+	isEmptyDefaultBlock,
+	isMovable,
+	isParentOfSelectedBlock,
+	isDraggable,
+	isSelectionEnabled,
+	className,
+	name,
+	isValid,
+	attributes,
+	initialPosition,
+	wrapperProps,
+	setAttributes,
+	onReplace,
+	onInsertBlocksAfter,
+	onMerge,
+	onSelect,
+	onRemove,
+	onInsertDefaultBlockAfter,
+	toggleSelection,
+	onShiftSelection,
+	onSelectionStart,
+} ) {
+	// Random state used to rerender the component if needed, ideally we don't need this
+	const [ , updateRerenderState ] = useState( {} );
+	const rerender = () => updateRerenderState( {} );
+
+	// Reference of the wrapper -------------------------------------------------
```

Comment:
What's with the dashes? I'd rather we embrace the pain of navigating "sections" of code to motivate us toward refactoring to separate functions/effects/etc, than to set some precedent for sectioning by code comments.